frontend/lib/DataModels/User/user.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:adpv_frontend/DataModels/User/auth_token_response.dart';
+import 'package:jwt_decode/jwt_decode.dart';
+
+enum UserRole {
+  user,
+  admin,
+  empty,
+}
+
+class User {
+  late AuthTokenResponse tokenResponse;
+  late String email;
+  List<UserRole> userRoles = [];
+
+  User(this.tokenResponse) {
+    final Map<String, dynamic> parsed = Jwt.parseJwt(tokenResponse.accessToken);
+    email = parsed['sub'];
+    for (var element in (parsed['roles'] as List)) {
+      userRoles.add(fromString(element));
+    }
+  }
+
+  UserRole fromString(String roleString) {
```

Comment:
```suggestion
  UserRole fromString(String roleString) {
    try{
      return UserRole.values.byName(roleString..toLowerCase());

  }catch(error){
    return UserRole.empty
  }
  }
```

Reply:
Imo tak jak Kuba zrobił jest dobre, bo jawnie mamy rozpisane przypadki, które mogą być i co robimy w przypadku braku znalezienia roli, ja jakoś osobiście nie jestem fanem wrzucania takich rzeczy w `try .. catch`

Reply:
oki, imo ładniej to wygląda, ale jak twoim zdaniem to jest mniej jasne to mozesz pominąc sugestie

Reply:
przyjdzie Ci jeszcze rola użytkownika technicznego, jednak wolę się z tym nie zgubić